fiftyone/core/session.py

```diff
@@ -617,33 +682,160 @@ def _display_colab(port, height):
     """
     import IPython.display
 
-    shell = """
-        (async () => {
-            const url = new URL(await google.colab.kernel.proxyPort(%PORT%, {'cache': true}));
-            url.searchParams.set('fiftyoneColab', 'true');
-            url.searchParams.set('notebook', 'true');
-            const iframe = document.createElement('iframe');
-            iframe.src = url;
-            iframe.setAttribute('width', '100%');
-            iframe.setAttribute('height', '%HEIGHT%');
-            iframe.setAttribute('frameborder', 0);
-            document.body.appendChild(iframe);
-        })();
-    """
-    replacements = [
-        ("%PORT%", "%d" % port),
-        ("%HEIGHT%", "%d" % height),
-    ]
-    for (k, v) in replacements:
-        shell = shell.replace(k, v)
-    script = IPython.display.Javascript(shell)
+    # pylint: disable=no-name-in-module,import-error
+    from google.colab import output
+
+    style_text = Template(_SCREENSHOT_STYLE).render(handle=uuid)
+    html = Template(_SCREENSHOT_COLAB).render(style=style_text, handle=uuid)
+    script = Template(_SCREENSHOT_COLAB_SCRIPT).render(
+        port=port, handle=uuid, height=height
+    )
 
-    IPython.display.display(script)
+    handle.display(IPython.display.HTML(html))
+    output.eval_js(script)
+
+    def capture(img):
+        idx = session._colab_img_counter[uuid]
+        session._colab_img_counter[uuid] = idx + 1
+        with output.redirect_to_element("#focontainer-%s" % uuid):
+            # pylint: disable=undefined-variable
+            display(
+                IPython.display.HTML(
+                    """
+                <img id='fo-%s%d' class='foimage' src='%s'/>
+                <style>
+                #fo-%s%d {
+                    display: none;
+                }
+                </style>
+                """
+                    % (uuid, idx, img, uuid, idx - 1)
+                )
+            )
 
+    output.register_callback("fiftyone.%s" % uuid.replace("-", "_"), capture)
 
-def _display_ipython(port, height):
+
+def _display_ipython(session, handle, uuid, port, height, update=False):
     import IPython.display
 
-    src = "http://localhost:%d/?notebook=true" % port
+    src = "http://localhost:%d/?notebook=true&handleId=%s" % (port, uuid)
     iframe = IPython.display.IFrame(src, height=height, width="100%")
-    IPython.display.display(iframe)
+    if update:
+        handle.update(iframe)
+    else:
+        handle.display(iframe)
+
+
+_SCREENSHOT_STYLE = """
```

Comment:
could consider moving screenshotting stuff to a separate module to avoid readers of `fiftyone.core.session` from getting intimidated (asking for a friend ;-)